### DISCUSSION TITLE
Cat-Printer 0.2.0

### DISCUSSION THREAD

**Discussion by NaitLee** [NaitLee/Cat-Printer#15] [Announcements]
It's a big update on backend.

Anyway, in all cases the printing process is much more reactive, because it will connect to printer just after scanning.

Also "smart" buffer sending! No need to care "frequency" anymore.

The typewriter hack *fully* works! Along with many developer goodies.

You can also flip the print direction, for example you want cat printer face you toward rather than downward, or print text with reversed order.

Apk is release version, signed, and slightly smaller.

<hr /><em>This discussion was created from the release <a href='https://github.com/NaitLee/Cat-Printer/releases/tag/0.2.0'>Cat-Printer 0.2.0</a>.</em>

**Comment by frankenstein91** [comment]
Hi Nait

I did some testing today... It did not work at first. But after buliding a bundle local and using chrome and not firefox I was able to print. CUPS did work after build and not from repo.
I do not know if that is normal.

We need an Option to make the black darker. I have some colored paper and it is hard to read.

**Comment by NaitLee** [comment]
I didn't find problem in any environment. I guess it's browser cache, but just guess. Error message can tell more...

The printer have "energy" concept, seems to make it darker, but seems useless on my GB02.
There's also "text" mode, I can't sure if this will help.
Currently, in `printer.py`, replace `self.image_mode()` with `self.text_mode()`, and try printing.

I slightly remember half year ago I accidentally made it much darker... but much are forgotten...
